tests/device_tests/cardano/test_address_public_key.py

```diff
@@ -0,0 +1,88 @@
+# This file is part of the Trezor project.
+#
+# Copyright (C) 2012-2019 SatoshiLabs and contributors
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# as published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+import pytest
+
+from trezorlib.cardano import create_address_parameters, get_address, get_public_key
+from trezorlib.messages import CardanoAddressType
+from trezorlib.tools import parse_path
+
+from ...common import parametrize_using_common_fixtures
+
+pytestmark = [
+    pytest.mark.altcoin,
+    pytest.mark.cardano,
+    pytest.mark.skip_t1,
+    pytest.mark.skip_ui,
+]
+
+ADDRESS_TYPES = {
+    "base": CardanoAddressType.BASE,
+    "base_script_key": CardanoAddressType.BASE_SCRIPT_KEY,
+    "base_key_script": CardanoAddressType.BASE_KEY_SCRIPT,
+    "base_script_script": CardanoAddressType.BASE_SCRIPT_SCRIPT,
+    "pointer": CardanoAddressType.POINTER,
+    "pointer_script": CardanoAddressType.POINTER_SCRIPT,
+    "enterprise": CardanoAddressType.ENTERPRISE,
+    "enterprise_script": CardanoAddressType.ENTERPRISE_SCRIPT,
+    "byron": CardanoAddressType.BYRON,
+    "reward": CardanoAddressType.REWARD,
+    "reward_script": CardanoAddressType.REWARD_SCRIPT,
+}
+
+
+@parametrize_using_common_fixtures(
+    "cardano/get_address_byron.json",
+    "cardano/get_address_byron.slip39.json",
+    "cardano/get_base_address.json",
+    "cardano/get_base_address_with_staking_key_hash.json",
+    "cardano/get_enterprise_address.json",
+    "cardano/get_pointer_address.json",
+    "cardano/get_reward_address.json",
+)
+def test_cardano_get_address(client, parameters, result):
+    staking_path = parameters.get("staking_path")
+    if staking_path is not None:
+        staking_path = parse_path(staking_path)
+    staking_key_path_hash = parameters.get("staking_key_hash")
+    if staking_key_path_hash is not None:
+        staking_key_path_hash = bytes.fromhex(staking_key_path_hash)
+    address = get_address(
+        client,
+        address_parameters=create_address_parameters(
+            address_type=ADDRESS_TYPES[parameters["address_type"]],
```

Comment:
let's cheat and do `address_type=getattr(CardanoAddressType, parameters["address_type"])`, instead of typing out all the types

Reply:
relevant: #1237

Reply:
:+1: We would need to add `.upper()` or do the uppercase in the fixtures. But I am fine with the first. What about you?

Reply:
I'd prefer uppercase in JSON but no strong feelings either way. Let's also ask Suite for how they handle enum values, maybe there's an answer there

Reply:
Ok, leaving with `upper()` for now. Done in bbf842d193e4660fef32b2afe12f6196164df878